reframe/core/pipeline.py

```diff
@@ -1057,6 +1060,9 @@ def run(self):
         commands = [*self.pre_run, ' '.join(exec_cmd), *self.post_run]
         environs = [self._current_partition.local_env,
                     self._current_environ, self._user_environ]
+        if self._extra_env:
+            environs.append(self._extra_env)
```

Comment:
Remove completely this part. Also the order of the environments should be the following: local_env, current_environ, container_env, user_environ.